src/discovery/composite.ts

```diff
@@ -11,8 +11,8 @@ export class CompositeDiscoveryNetwork implements IDiscoveryNetwork {
         }
     }
 
-    public async ping() {
-        await Promise.all(this.delegates.map(d => d.ping()));
+    public async ping(deviceIp: string | undefined) {
```

Comment:
Let's go with 

```suggestion
    public async ping(deviceIp?: string) {
```

since the param may be omitted.